lib/neo4j-server/cypher_session.rb

```diff
@@ -7,7 +8,22 @@ module Server
     end
 
     class CypherSession < Neo4j::Session
+      module PrivateMethods
+        private
+
+        def extract_faraday_options(params, defaults = {})
+          verify_faraday_options(params.delete(:faraday_options) || params.delete('faraday_options') || {}, defaults)
+        end
+
+        def extract_basic_auth(url, params)
+          return unless url && URI(url).userinfo
```

Comment:
Not a huge deal, but you could perhaps create a `uri` object at the start and use it in all of the places you're using `URI(url)`